jupyterhub/services/auth.py

```diff
@@ -138,14 +139,46 @@ class HubAuth(Configurable):
     def _cookie_cache(self):
         return _ExpiringDict(self.cookie_cache_max_age)
 
-    def _check_response_for_authorization(self, r):
-        """Verify the response for authorizing a user
+    def _check_hub_authorization(self, url, cache_key=None, use_cache=True):
+        """Identify a user with the Hub
+        
+        Args:
+            url (str): The API URL to check the Hub for authorization
+                       (e.g. http://127.0.0.1:8081/hub/api/authorizations/token/abc-def)
+            cache_key (str): The for checking the cache
+            use_cache (bool): Specify use_cache=False to skip cached cookie values (default: True)
+
+        Returns:
+            user_model (dict): The user model, if a user is identified, None if authentication fails.
+        An HTTPError is raised if 
+        Verify the response for authorizing a user
 
         Raises an error if the response failed, otherwise returns the json
         """
+        if use_cache:
+            if cache_key is None:
+                raise ValueError("cache_key is required when using cache")
+            cached = self.cookie_cache.get(cache_key)
+            if cached is not None:
+                return cached
```

Comment:
`cached` is the cached result of this method, from calling this method before with the same argument. If a cached result is found, return it instead of running the method again. It does not continue after the `return`.